aws/resource_aws_globalaccelerator_listener.go

```diff
@@ -0,0 +1,270 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/globalaccelerator"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsGlobalAcceleratorListener() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGlobalAcceleratorListenerCreate,
+		Read:   resourceAwsGlobalAcceleratorListenerRead,
+		Update: resourceAwsGlobalAcceleratorListenerUpdate,
+		Delete: resourceAwsGlobalAcceleratorListenerDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"accelerator_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"client_affinity": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
```

Comment:
We should switch this to `Default: globalaccelerator.ClientAffinityNone,` to match the [API default](https://docs.aws.amazon.com/global-accelerator/latest/api/API_CreateListener.html) and so we can properly perform drift detection when unconfigured.

```suggestion
				Default: globalaccelerator.ClientAffinityNone,
```